cargo-dist/src/lib.rs

```diff
@@ -149,9 +150,44 @@ fn run_build_step(
             dist_graph, committish, prefix, target,
         )?),
         BuildStep::Extra(target) => run_extra_artifacts_build(dist_graph, target),
+        BuildStep::Updater(updater) => fetch_updater(dist_graph, updater),
     }
 }
 
+/// Fetches an installer executable and installs it in the expected target path.
+pub fn fetch_updater(dist_graph: &DistGraph, updater: &UpdaterStep) -> Result<()> {
+    let tmpdir = TempDir::new().into_diagnostic()?;
+
+    // Update this to work from releases, and to fetch prebuilt binaries,
+    // once this has been released at least once.
+    let mut cmd = Cmd::new(
+        "cargo",
+        format!("Fetch installer for {}", updater.target_triple),
+    );
+
+    // Install to a temporary path before moving it to the destination
+    cmd.arg("install")
+        .arg("--root")
+        .arg(tmpdir.path())
+        .arg("--git")
+        .arg("https://github.com/axodotdev/axoupdater")
+        .arg("--all-features")
+        .arg("--bin")
+        .arg("axoupdater");
+
+    cmd.run()?;
+
+    // OK, now we have a binary in the tempdir
+    let mut source = tmpdir.path().join("bin").join("axoupdater");
+    if updater.target_triple.contains("windows") {
+        source.set_extension("exe");
+    }
```

Comment:
i'm fairly certain we have this kind of logic somewhere else, and should reuse/factor it.

Reply:
There is another place, but it operates on strings of base filenames: https://github.com/axodotdev/cargo-dist/blob/main/cargo-dist/src/tasks.rs#L1150-L1151